packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -65,10 +65,22 @@ abstract class DirectionalCaretMovementIntent extends DirectionalTextEditingInte
   const DirectionalCaretMovementIntent(
     bool forward,
     this.collapseSelection,
-    [this.collapseAtReversal = false]
+    [
+      this.collapseAtReversal = false,
+      this.expand = false,
+    ]
   ) : assert(!collapseSelection || !collapseAtReversal),
       super(forward);
 
+  /// Whether to expand an uncollapsed selection or extend it.
+  ///
+  /// When set to false, the selection will extend, meaning that the base will
+  /// be held in place while the extent is moved.
+  ///
+  /// When set to true, the selection will expand, meaning that the selection
+  /// will grow by moving either the base or the extent.
```

Comment:
Also maybe say this is typically only for macOS?

Reply:
This exact comment was removed because I created a separate Intent.  I made sure that Intent's comments were clear and I added a line about MacOS there.